libraries/chain/transaction.cpp

```diff
@@ -255,6 +255,13 @@ vector<bytes> packed_transaction::get_context_free_data()const
    } FC_CAPTURE_AND_RETHROW((compression)(packed_context_free_data))
 }
 
+transaction_id_type packed_transaction::id()const
```

Comment:
I was looking into this function because I was concerned with potential performance impact (it unpacks the transaction just to get its id) and couldn't find any place it was used. Do we want to add it if we don't need it?

Reply:
This is intended to be used inside the net plugin for handling packed transactions and weeding out duplicates before calling the chain controller. That way we avoid all the error handling logic. It seems that my git stashing, diffing, patching, and reverting managed to whack that code. I'm reapplying the change now.

Reply:
Regarding the performance impact potential, that is why I added a function to the packed transaction type - this function can be refactored to, for example, if a separate unpacked transaction id is carried with the packed transaction. Or perhaps the intermediate unpacked transaction could be cached so it doesn't need to be redone.